### PR TITLE
store address in StopPoint, and set these address during Data::complete()

### PR DESCRIPTION
**Pull request by pbench**
and use the stored addresses while filling protobuf responses

### CODE REVIEW

File: source/type/pt_data.cpp [hunk 1]
```diff
@@ -491,6 +492,14 @@ const StopPointConnection* PT_Data::get_stop_point_connection(const StopPoint& f
     return *search;
 }
 
+void PT_Data::fill_stop_point_address(PbCreator* pb_creator) {
+    for (StopPoint* stop_point : stop_points) {
+        const nt::GeographicalCoord* coord = &(stop_point->coord);
+        pbnavitia::Address* address = &(stop_point->address);
+        pb_creator->fill(coord, address, 3);
```
Comment (pbench) [inline source/type/pt_data.cpp:499]: is depth=3 right ?

File: source/ed/CMakeLists.txt [hunk 2]
```diff
@@ -5,7 +5,7 @@ SET(SOURCE_LIB
 )
 
 add_library(ed ${SOURCE_LIB})
-target_link_libraries(ed types fare)
+target_link_libraries(ed types fare data)
```
Comment (xlqian) [inline source/ed/CMakeLists.txt:8]: why `ed` is also involved?

File: source/type/pb_converter.cpp [hunk 3]
```diff
@@ -618,9 +618,8 @@ void PbCreator::Filler::fill_pb_object(const nt::StopPoint* sp, pbnavitia::StopP
         fill(sp->admin_list, stop_point->mutable_administrative_regions());
     }
 
-    if (depth > 2) {
-        fill(&sp->coord, stop_point);
-    }
+    *(stop_point->mutable_address()) = sp->address;
```
Comment (xlqian) [inline source/type/pb_converter.cpp:621]: fill(stop_point->mutable_address(), stop_point)

File: source/type/stop_point.h [hunk 4]
```diff
@@ -56,11 +57,12 @@ struct StopPoint : public Header, Nameable, hasProperties, HasMessages {
     std::vector<StopPointConnection*> stop_point_connection_list;
     std::set<Dataset*> dataset_list;
     boost::container::flat_set<Route*> route_list;
+    pbnavitia::Address address;
```
Comment (xlqian) [inline source/type/stop_point.h:60]: as @azime pointed out, in principle, we should create a new struct for address and fill the `pbnavitia:Address` in pb_converter... just like `type::Admin`

File: source/ed/CMakeLists.txt (hunk 2 above)
Comment (pbench) [reply source/ed/CMakeLists.txt:8]: what happens is that I store the protobuf Address in the StopPoint object (maybe not the best thing to do), consequently `types` depends on the protobuf files, and this impact dependencies everywhere :crying_cat_face:

**Review by xlqian — COMMENTED** [review]
:)

**Pull request closed without merging**